src/Mod/Draft/draftutils/gui_utils.py

```diff
@@ -129,13 +129,19 @@ def autogroup(obj):
         if Gui.ActiveDocument.ActiveView.getActiveObject("Arch"):
             # add object to active Arch Container
             active_arch_obj = Gui.ActiveDocument.ActiveView.getActiveObject("Arch")
+            if obj in active_arch_obj.InListRecursive:
+                # Prevents cyclic references and avoid autogroup in obj points to active_arch_obj
+                return
             active_arch_obj.addObject(obj)
 
         elif Gui.ActiveDocument.ActiveView.getActiveObject("part", False) is not None:
```

Comment:
I think `is not None` can be removed.

Reply:
I'm not sure... being it beyond the scope of this PR i'd prefer to leave it like this.